snuba/datasets/all_events.py

```diff
@@ -0,0 +1,165 @@
+from abc import ABC
+from datetime import timedelta
+from itertools import chain
+from typing import Any, Mapping, MutableSequence, Sequence
+
+from snuba.datasets.dataset import TimeSeriesDataset
+from snuba.datasets.dataset_schemas import DatasetSchemas
+from snuba.datasets.factory import get_dataset
+from snuba.datasets.schemas import Schema, RelationalSource
+from snuba.query.extensions import QueryExtension
+from snuba.query.parsing import ParsingContext
+from snuba.query.project_extension import ProjectExtension, ProjectWithGroupsProcessor
+from snuba.query.query import Query
+from snuba.query.query_processor import QueryProcessor
+from snuba.query.timeseries import TimeSeriesExtension
+from snuba.request.request_settings import RequestSettings
+from snuba.util import columns_in_expr, is_condition
+
+
+EVENTS = 'events'
+TRANSACTIONS = 'transactions'
+TRANSACTIONS_ONLY_COLUMNS = [
+    'trace_id',
+    'span_id',
+    'transaction_name',
+    'transaction_hash',
+    'transaction_op',
+    'start_ts',
+    'start_ms',
+    'finish__ts',
+    'finish_ms',
+    'duration',
+]
+
+EVENTS_ONLY_COLUMNS = [
+    'group_id',
+    'transaction',
+]
+
+
+# Based on query.get_all_referenced_columns()
+def get_condition_columns(query: Query) -> Sequence[Any]:
+    col_exprs: MutableSequence[Any] = []
+    # Conditions need flattening as they can be nested as AND/OR
+    if query.get_conditions():
+        flat_conditions = list(
+            chain(*[[c] if is_condition(c) else c for c in query.get_conditions()]))
+        col_exprs.extend([c[0] for c in flat_conditions])
+
+    # Return the set of all columns referenced in any expression
+    return set(chain(*[columns_in_expr(ex) for ex in col_exprs]))
```

Comment:
Feel free to make changes to the Query api to make get_all_referenced_columns more flexible. Something like this will be useful to others as well. (This does not mean "just copy this method back into Query". Separating the logic that iterates over columns and returns them in its own method and adding public methods that iterate over specific parts of the query (columns, conditions, etc.) seems a more flexible solution with little code duplication.